atc/engine/builder/step_factory.go

```diff
@@ -69,7 +69,7 @@ func (factory *stepFactory) GetStep(
 		factory.client,
 	)
 
-	return exec.LogError(getStep, delegate)
+	return exec.RetryError(getStep, delegate)
```

Comment:
On https://github.com/concourse/concourse/issues/5086#issuecomment-606018381 it was recommended to not have the retry functionality turned on by default. That would mean having some sort of check here as well as allowing users to specify a flag to turn this feature on.